src/pypi_attestation_models/_cli.py

```diff
@@ -0,0 +1,302 @@
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    inspect_command = subcommands.add_parser(
+        name="inspect",
+        help="Inspect one or more inputs",
+        parents=[parent_parser],
+    )
+
+    inspect_command.add_argument(
+        "--dump-bytes",
+        action="store_true",
+        default=False,
+        help="Dump the bytes of the signature",
+    )
+
+    inspect_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    return parser
+
+
+def _die(args: argparse.Namespace, message: str) -> NoReturn:
+    """Handle argument parsing errors and terminate the program.
+
+    Fix up the type hints on our use of `ArgumentParser.error`.
+    """
+    args._parser.error(message)  # noqa: SLF001.
+    raise ValueError("unreachable")
+
+
+def get_identity_token(args: argparse.Namespace) -> IdentityToken:
+    """Generate an Identity Token.
+
+    This method uses the following order of precedence:
+    - A token passed as an argument
+    - An ambient credential
+    - An OAuth-2 flow
+    """
+    # First, check if a token was supplied
+    if args.identity_token:
+        return IdentityToken(args.identity_token)
+
+    # Ambient credential detection
+    oidc_token = detect_credential()
+    if oidc_token is not None:
+        return IdentityToken(oidc_token)
+
+    # Finally, OAuth-2 Flow
+    if args.staging:
+        issuer = Issuer.staging()
+    else:
+        issuer = Issuer.production()
+
+    return issuer.identity_token()
+
+
+def _sign(args: argparse.Namespace) -> None:
+    """Sign the files passed as argument."""
+    try:
+        identity = get_identity_token(args)
+    except IdentityError as identity_error:
+        _die(args, f"Failed to detect identity: {identity_error}")
+
+    if args.staging:
+        signing_ctx = SigningContext.staging()
+    else:
+        signing_ctx = SigningContext.production()
+
+    with signing_ctx.signer(identity, cache=True) as signer:
+        for file_path in args.files:
+            _logger.debug(f"Signing {file_path}")
+
+            if not file_path.is_file():
+                _die(args, f"{file_path} is not a file.")
+
+            signature_path = Path(f"{file_path}.publish.attestation")
+            if signature_path.is_file():
+                _die(args, f"Signature already exists for {file_path}")
+
+            attestation = Attestation.sign(signer, file_path)
+            _logger.debug("Attestation saved for %s saved in %s", file_path, signature_path)
+
+            signature_path.write_text(attestation.model_dump_json())
+
+
+def _inspect(args: argparse.Namespace) -> None:
+    """Inspect attestations.
+
+    Warning: The information displayed from the attestations are not verified.
+    """
+    for file_path in args.files:
+        if not file_path.is_file():
+            _die(args, f"{file_path} is not a file.")
```

Comment:
Same for these (fail early if any of the files do not exist, before starting to process)